scene/resources/material.h

```diff
@@ -339,6 +340,9 @@ class BaseMaterial3D : public Material {
 		uint32_t feature_mask;
 		uint32_t flags;
 
+		// shader template
+		uint64_t shader_template_id;
```

Comment:
This may be a breaking change, I am not sure what the implications of changing the material key are. However seeing shader templates effect the compiled shader, it has to be part of the key.